integration/ctr_software_updatable_test.go

```diff
@@ -0,0 +1,372 @@
+					if isRemoving && isInstalledDeps && isRemoved {
+						return true, nil
+					}
+					return true, fmt.Errorf("container status is not expected")
+				}
+				return true, nil
+			} else if event.Path == fmt.Sprintf("/features/%s/properties/status/installedDependencies", suFeatureID) {
+				if isRemoving {
+					isInstalledDeps = true
+					return false, nil
+				}
+				return true, nil
+			} else if event.Path == fmt.Sprintf("/features/%s/properties/status/lastFailedOperation", suFeatureID) {
+				eventValue, err = parseMap(event.Value)
+				require.NoError(suite.T(), err, "failed to parse event value")
+
+				propertyMessage, err := parseString(eventValue["message"])
+				require.NoError(suite.T(), err, "failed to parse property message")
+				return true, fmt.Errorf(propertyMessage)
+			}
+
+			return true, fmt.Errorf("event for an unexpected container status while removing is received")
+		}
+		return true, fmt.Errorf("unknown message is received")
+	})
+	if err != nil {
+		wsConnection.Close()
+	}
+	return err
+}
+
+func (suite *softwareUpdatableSuite) TestSoftwareInstallRemove() {
+	ctrID, err := suite.installContainer(suite.createParameters(false))
+	require.NoError(suite.T(), err, "failed to process installing the container")
+
+	err = suite.removeContainer(createRemoveParams(ctrID))
+	require.NoError(suite.T(), err, "failed to process removing the container")
+}
+
+func (suite *softwareUpdatableSuite) TestSoftwareInstallWithInvalidParameters() {
+	wsConnection := suite.createWSConnection()
+	defer suite.closeUnsubscribe(wsConnection)
+
+	params := make(map[string]interface{})
+	_, err := util.ExecuteOperation(suite.Cfg, suite.swUpdatableFeatureURL, actionInstall, params)
+	if err != nil {
+		wsConnection.Close()
+	}
+	require.Errorf(suite.T(), err, "failed to execute software updatable install for containers with params %v", params)
+
+}
+
+func (suite *softwareUpdatableSuite) TestSoftwareInstallWithWrongChecksum() {
+	_, err := suite.installContainer(suite.createParameters(true))
+	require.ErrorContains(suite.T(), err, "internal runtime error")
+}
+
+func (suite *softwareUpdatableSuite) TestSoftwareRemoveNonExistingContainer() {
+	ctrID := "NonExistingContainer"
+	err := suite.removeContainer(createRemoveParams(ctrID))
+	require.ErrorContains(suite.T(), err, "container with ID = "+ctrID+" does not exist")
+}
+
+func (suite *softwareUpdatableSuite) createParameters(wrongChecksum bool) map[string]interface{} {
+	src, err := download(validJSONUrl)
+	require.NoError(suite.T(), err, "unable to download file from url "+validJSONUrl)
+	splitStr := strings.Split(validJSONUrl, "/")
+	filePath := "/tmp/" + splitStr[len(splitStr)-1]
+	err = os.WriteFile(filePath, src, 7777)
+	require.NoError(suite.T(), err, "unable to write file with path "+filePath)
+
+	params := createInstallParams(filePath, src, wrongChecksum)
+	err = os.Remove(filePath)
+	require.NoError(suite.T(), err, "unable to remove file with path "+filePath)
+	return params
```

Comment:
could you try this way?
```suggestion
	require.NoError(suite.T(), os.WriteFile(filePath, src, 7777), "unable to write file with path "+filePath)
	defer require.NoError(suite.T(),  os.Remove(filePath), "unable to remove file with path "+filePath)

	return createInstallParams(filePath, src, wrongChecksum)
```